src/lib.rs

```diff
@@ -83,17 +83,42 @@ pub mod scope;
 #[cfg(feature = "tracy")]
 pub mod tracy;
 
+/// The result of a gpu timer scope.
 pub struct GpuTimerScopeResult {
+    /// Label that was specified when opening the scope.
     pub label: String,
+
     /// Time range of this scope in seconds.
+    ///
+    /// None if the scope was not profiled.
+    /// This if profiling for the kind of scope was disabled.
+    ///
     /// Meaning of absolute value is not defined.
-    pub time: Range<f64>,
+    pub time: Option<Range<f64>>,
 
+    /// Scopes that were opened while this scope was open.
     pub nested_scopes: Vec<GpuTimerScopeResult>,
     pub pid: u32,
     pub tid: ThreadId,
 }
 
+/// A wgpu-profiler error.
+#[derive(thiserror::Error, Debug, PartialEq, Eq)]
+pub enum GpuProfilerError {
+    #[error("All profiling scopes need to be closed before ending a frame. The following scopes were still open: {0:?}")]
+    UnclosedScopesAtFrameEnd(Vec<String>),
+
+    #[error(
+        "Not all queries were resolved before ending a frame.\n
+Call `GpuProfiler::resolve_queries` after all profiling scopes have been closed and before ending the frame.\n
+There were still {0} queries unresolved"
+    )]
+    UnresolvedQueriesAtFrameEnd(u32),
+
+    #[error("No profiler GpuProfiler scope was previously opened. For each call to `end_scope` you first need to call `begin_scope`.")]
+    NoOpenScope,
+}
```

Comment:
It looks like the first two can be emitted by `end_frame` and the last one by `end_scope`. Since these don't overlap, it seems like it would be easy and preferable to have two separate error types.

Reply:
Huh, yeah I found it cumbersome, but you're right why not split them